pkg/batchfetcher/batchfetcher.go

```diff
@@ -104,16 +106,12 @@ func NewModule(mc *ModuleConfig, epochNr t.EpochNr, clientProgress *clientprogre
 
 	// The AppRestoreState handler restores the batch fetcher's state from a checkpoint
 	// and forwards the event to the application, so it can restore its state too.
-	eventpbdsl.UponAppRestoreState(m, func(restoreState *eventpb.AppRestoreState) error {
-
-		// Load the checkpoint from the received event.
-		chkp := checkpoint.StableCheckpointFromPb(restoreState.Checkpoint)
```

Comment:
I'd actually prefer using the checkpoint abstraction. It might be marginally les performant, but I find it cleaner.

Reply:
Done as requested, but may come back to this once I pass through the checkpoint module. Bear in mind that `checkpoint.` refers to a manually written package, whereas `checkpointpbtypes.` is automatically generated. I believe minimizing the amount of manually written code and maximizing tags in protobufs is the way to go. If types change later at least we know that they must be changed just in protobufs.